contentcuration/contentcuration/views/public.py

```diff
@@ -107,3 +108,10 @@ def list(self, request):
                 'operating_system': None,
                 }
         return Response(info)
+
+
+@api_view(['GET'])
+@permission_classes((AllowAny,))
+def get_catalog_details(request, item_id):
```

Comment:
I think the backend implementation here could be simplified significantly, by just adding pagination to the Channel Viewset that is already defined.

For the `download_count` field, I am a bit iffy about displaying publicly, because I don't know that it's a stat that we can say is completely meaningful without a lot of filtering.

This particular endpoint appears to be equivalent to doing a detail view request on the viewset, but using the channel serializer for read behaviour, which the viewset is intended to avoid. Probably cleaner to avoid adding new public API endpoints until we have a clear need for them.